### PR TITLE
Added support for Gradle authenticated repositories

### PR DESCRIPTION
**Pull request by GeorgiosGoniotakis**
Hello there 😄 Thank you for creating Dependabot. As discussed [here](https://github.com/dependabot/dependabot-core/issues/1232), using the Maven module as a guide, we enhanced the capabilities of the Gradle module. The module now supports retrieval of dependencies and plugins from authenticated Gradle repositories. Any changes to the code have been performed using the conventions utilized in dependabot-core and additional tests which exercise the new functionality have been included. To test the new functionality, please include additional credentials as follows:

```ruby
$options[:credentials] << {
  "type" => "maven_repository",
  "url" => "URL_OF_PRIVATE_REPO_HERE",
  "username" => "USERNAME_HERE",
  "password" => "PASSWORD_HERE"
}
```
For any recommendations or queries about our contribution, please feel free to reach out to us. Happy coding 😄 

Closes https://github.com/dependabot/feedback/issues/755

### CODE REVIEW

**Comment by GeorgiosGoniotakis** [comment]
Hello @rebelagentm and @feelepxyz and happy new year. Are there any news regarding this?

File: gradle/lib/dependabot/gradle/update_checker/version_finder.rb [hunk 1]
```diff
@@ -160,47 +167,97 @@ def google_version_details
...
-              raise if repository_url == central
+              raise if central_repo_urls.include?(repository_details["url"])
 
               Nokogiri::XML("")
             end
         end
 
         def repository_urls
-          plugin? ? plugin_repository_urls : dependency_repository_urls
+          plugin? ? plugin_repository_urls : dependency_repository_details
```
Comment (rebelagentm) [inline gradle/lib/dependabot/gradle/update_checker/version_finder.rb]:
```suggestion
          plugin? ? plugin_repository_details : dependency_repository_details
```

File: gradle/lib/dependabot/gradle/update_checker/version_finder.rb [hunk 2]
```diff
@@ -160,47 +167,97 @@ def google_version_details
...
         end
 
-        def plugin_repository_urls
-          [GRADLE_PLUGINS_REPO] + dependency_repository_urls
+        def plugin_repository_details
+          [{
+            "url" => GRADLE_PLUGINS_REPO,
+            "username" => nil,
+            "password" => nil
+          }]
```
Comment (rebelagentm) [inline gradle/lib/dependabot/gradle/update_checker/version_finder.rb]:
```suggestion
          }] + dependency_repository_details
```

Comment (rebelagentm) [reply gradle/lib/dependabot/gradle/update_checker/version_finder.rb]: It looks like we used to append the URLs here.  I think we would still want to add them to the end of our new hash.

**Comment by SkyeWelch** [comment]
Would you like a hand with any of these changes, to get them over the line, @GeorgiosGoniotakis?

**Comment by GeorgiosGoniotakis** [comment]
Hello @SkyeWelch The last notification by @rebelagentm went to my email folder which I use for GitHub messages. Thank you for offering to help.

**Comment by SkyeWelch** [comment]
No problem! Thank you for putting this PR up, and shout if you do want a hand with it in the future.

**Comment by GeorgiosGoniotakis** [comment]
Hello @rebelagentm I made all the necessary changes but it seems that Circle CI is failing with an unclear error. Could you please take a look?

**Comment by feelepxyz** [comment]
@GeorgiosGoniotakis might be old codes, could you try rebasing from master?

**Comment by GeorgiosGoniotakis** [comment]
The rebase that I made messed up things quite a lot. I will open a new PR with my changes.

**Pull request closed without merging**